exosphere-shared/src/docker-helper.ls

```diff
@@ -1,146 +1,49 @@
 require! {
-  \child_process
   \prelude-ls : {any}
-  \observable-process : ObservableProcess
-  \os
+  'dockerode' : Docker
+  'wait' : {wait}
 }
 
+docker = new Docker
 
+# Helper class used to manage Docker processes not started by docker-compose
 class DockerHelper
 
-  @build-all-images = ({write}, cb) ->
-    new ObservableProcess('docker-compose build'
-                          stdout: {write}
-                          stderr: {write})
-      ..on 'ended', cb
+  @start-container = ({Image, name, HostConfig}, done) ->
+    DockerHelper._container-is-running name, (err, is-running) ->
+      | err => done err
+      | is-running => done!
+      | otherwise =>
+        docker.create-container {Image, name, HostConfig}, (err, container) -> 
+          | err => done err
+          container.start (err, data) ->
+            | err => done err
+            wait 2_000, done
 
-  @pull-all-images = ({write}, cb) ->
-    new ObservableProcess('docker-compose pull'
-                          stdout: {write}
-                          stderr: {write})
-      ..on 'ended', cb
 
+  @remove-container = ({name}, done) ->
+    docker.list-containers {name}, (err, containers) ->
+      | err => done err
+      DockerHelper._force-remove-containers containers, done
```

Comment:
`@remove-container` additionally passes a `{name}` to `docker.list-containers`, meaning that only containers matching the name will be passed, and therefore removed in the `_force-remove-containers` call. 

However, I think these two could be combined by having a default value for `name` of an empty string?
```
@remove-containers = ({name = ''}, done) -> 
    docker.list-containers {name}, (err, containers) ->
        | err                 => done err
        | !containers.length  => done!
        | _                   => DockerHelper._force-remove-containers containers, done
```